wagtail/documents/views/documents.py

```diff
@@ -1,131 +1,154 @@
+    context_object_name = "documents"
+    page_title = gettext_lazy("Documents")
+    header_icon = "doc-full-inverse"
+    page_kwarg = "p"
+    paginate_by = 20
+    index_url_name = "wagtaildocs:index"
+    index_results_url_name = "wagtaildocs:index_results"
+    add_url_name = "wagtaildocs:add_multiple"
+    edit_url_name = "wagtaildocs:edit"
+    template_name = "wagtaildocs/documents/index.html"
+    results_template_name = "wagtaildocs/documents/index_results.html"
+    default_ordering = "title"
+    table_class = DocumentTable
+    model = get_document_model()
+    add_item_label = gettext_lazy("Add a document")
+    show_other_searches = True
+
+    def get_base_queryset(self):
         # Get documents (filtered by user permission)
-        documents = permission_policy.instances_user_has_any_permission_for(
+        return self.permission_policy.instances_user_has_any_permission_for(
             self.request.user, ["change", "delete"]
-        )
-
-        # Ordering
-        if "ordering" in self.request.GET and self.request.GET["ordering"] in [
-            "title",
-            "-created_at",
-        ]:
-            ordering = self.request.GET["ordering"]
-        else:
-            ordering = "-created_at"
-        documents = documents.order_by(ordering)
+        ).select_related("collection")
 
-        # Filter by collection
+    def filter_queryset(self, queryset):
         self.current_collection = None
         collection_id = self.request.GET.get("collection_id")
         if collection_id:
             try:
                 self.current_collection = Collection.objects.get(id=collection_id)
-                documents = documents.filter(collection=self.current_collection)
+                queryset = queryset.filter(collection=self.current_collection)
             except (ValueError, Collection.DoesNotExist):
                 pass
 
-        # Search
-        query_string = None
-        if "q" in self.request.GET:
-            self.form = SearchForm(self.request.GET, placeholder=_("Search documents"))
-            if self.form.is_valid():
-                query_string = self.form.cleaned_data["q"]
-                if query_string:
-                    search_backend = get_search_backend()
-                    documents = search_backend.autocomplete(query_string, documents)
-        else:
-            self.form = SearchForm(placeholder=_("Search documents"))
+        return self.filters, queryset
+
+    @cached_property
+    def columns(self):
+        columns = [
+            BulkActionsColumn("bulk_actions"),
+            TitleColumn(
+                "title",
+                label=_("Title"),
+                sort_key="title",
+                get_url=self.get_edit_url,
+                get_title_id=lambda doc: f"document_{quote(doc.pk)}_title",
+            ),
+            DownloadColumn("filename", label=_("File")),
+            DateColumn("created_at", label=_("Created"), sort_key="created_at"),
```

Comment:
Super tiny nitpick: these column definitions don't preserve the widths from the original template.
```suggestion
            BulkActionsColumn("bulk_actions", width="10px"),
            TitleColumn(
                "title",
                label=_("Title"),
                sort_key="title",
                get_url=self.get_edit_url,
                get_title_id=lambda doc: f"document_{quote(doc.pk)}_title",
            ),
            DownloadColumn("filename", label=_("File")),
            DateColumn("created_at", label=_("Created"), sort_key="created_at", width="16%"),
```

Reply:
Thanks! Restored in 2bd273d and merged.